### DISCUSSION TITLE
Question about writing application against css branch

### DISCUSSION THREAD

**Discussion by erinxocon** [Textualize/textual#267] [Q&A] [answered]
Hello! 

Great project! I was exploring it tonight and I wanted to try out some of the css stuff. I saw that active development was going on via the `css` branch. I checked it out and did a `poetry install`. I tried to write some of my own sample apps and also tried to run the sample code. I can't seem to bind to any keys. I get the following message anytime I run code that tries to await a bind. 

```
root@845b9cd33163:/source# poetry run python examples/animation.py 
╭────────────────────────────────────────────────────── Traceback (most recent call last) ───────────────────────────────────────────────────────╮
│                                                                                                                                                │
│ /source/examples/animation.py:11 in on_load                                                                                                    │
│                                                                                                                                                │
│    8 │                                                                                       ╭────────────── locals ──────────────╮            │
│    9 │   async def on_load(self, event) -> None:                                             │ event = Load()                     │            │
│   10 │   │   """Bind keys here."""                                                           │  self = SmoothApp(title='Textual') │            │
│ ❱ 11 │   │   await self.bind("b", "toggle_sidebar", "Toggle sidebar")                        ╰────────────────────────────────────╯            │
│   12 │   │   await self.bind("q", "quit", "Quit")                                                                                              │
│   13 │                                                                                                                                         │
│   14 │   show_bar = Reactive(False)                                                                                                            │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: object NoneType can't be used in 'await' expression
```

If I have the main branch checked out, everything works as intended. I am not opening up an issue about this as 1) I know it is in active development so the api could have changed. and 2) I assume the dev's can develop on the branch and run tests and apps so I'm curious as to why I can't! I did see [this closed issue](https://github.com/Textualize/textual/issues/66) with the same error message but it was not really resolved.

My environment if it matters:

```
OS: macOS 12.2 21D49 arm64
Host: MacBookPro18,3
Kernel: 21.3.0
Terminal: iTerm2 
```

I also tried this inside of a docker to see if it could have been a mac vs linux thing, however not knowing much about how curses and terminals work, this might not be a great test. 

Thanks for any advice! This looks like a very cool project and I have some ideas and would love to experiment!

-Erin

**Comment by ThatXliner** [comment, marked as answer]
It appears that the `bind` method on `App` is no longer an asynchronous method on the `css` branch (along with a [few other things](https://github.com/Textualize/textual/commit/ad8f7e893a5f4698ba4b946e437e870f415bb894))

https://github.com/Textualize/textual/blob/06417bb9b692b5466545371f096ddf51946a2761/src/textual/app.py#L188

But on the master branch (as I am writing this), it's still an asynchronous method

https://github.com/Textualize/textual/blob/4521ecb66a494aab9b68ada9ba7db8f7a4cefde0/src/textual/app.py#L165

**Reply by erinxocon** [reply]
🤦‍♀️ This morning looking at the code, this seems very obvious! I appreciate you shining a light on what should have been obvious!!

**Comment by willmcgugan** [comment]
@erinxocon The examples apps are currently broken, with the exception of basic.py I think. But I'm afraid there will a lot of instability until we work through the changes to the API.

**Reply by erinxocon** [reply]
@willmcgugan I  appreciate it. I figured it was due to API instability with the cutting edge stuff. @ThatXliner pointed out the api change above. I'm still learning the code base and couldn't use my normal debugging tools so I wasn't sure where the bind method was defined, thought I'd ask here while I'm learning 😅 Thanks for the great work on these projects! Sorry for the bother.

**Reply by willmcgugan** [reply]
@erinxocon No bother. Here to help :)

**Reply by erinxocon** [reply]
@willmcgugan if I fixed the examples and got them working again with the css branch would that be something you're looking for right now? If so I'd be happy to submit a PR! If you are going to completely redo the samples than perhaps it's not worth the effort.

**Reply by willmcgugan** [reply]
Thanks, but we will be redoing most of the examples. They’re going to get even simpler! Glad you got it working though. Let me know how you get on.

**Reply by delulu3i4** [reply]
@willmcgugan sorry if answer this question before but do you have any high level milestones/timelines thinking? Nothing of concern more just a curiosity?
